stdlib/bytes.mli

```diff
@@ -463,6 +463,91 @@ val of_seq : char Seq.t -> t
 (** Create a string from the generator
     @since 4.07 *)
 
+(** {6 Binary storage of numbers} *)
+
+external get_int16_ne : bytes -> int -> int = "%caml_bytes_get16"
```

Comment:
IIRC there's no advantage to expose this as an external at the `.mli` level.

Reply:
Exposing the external in the `.mli` (and hence in the `.cmi`) allows every use of this function to compile a call to the primitive directly (no passing through a wrapper function). Same as inlining, but this works even if inlining does not take place (e.g. in bytecode).

Reply:
Is this an important property, though? Because the cost is that if we ever decide to change the implementation (for example wrapping it into a bound-checking or exception-changing routine) or primitive name, we break the interface, and that seems quite high to me.

Reply:
I am not sure I understand: is there a way to write code that would break if one changes the signature from `external` to `val` ?

Reply:
Some people like to copy the interfaces from the stdlib for their own implementations, and they would get bitten by this.

Other people avoid this problem by `include`ing the interfaces, and they get bitten every time we add a function or even an optional parameter to an existing function.

So I would like to see the `external`s removed from the .mlis. Bytecode speed is not an issue nowadays.

Reply:
I guess @gasche was referring to something like:

```
module F (X : sig
    external f : unit -> unit = "%ignore"
end) = struct
end

module OK = F (struct
    external f : unit -> unit = "%ignore"
  end)

module KO = F (struct
    let f () = ()
  end)
```